tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/demo/tf-interactive-inference-age-demo.html

```diff
@@ -178,60 +188,100 @@
+                'workclass': 11
+              };
+              const xMin = +e.detail.x_min;
+              const xMax = +e.detail.x_max;
+              const isNum = !this.$.dash.partialDepPlotEligibleFeatures[
+                featureMapping[e.detail.feature_name]].samples;
+              const numTrials = isNum ? 10 :
+                this.$.dash.partialDepPlotEligibleFeatures[featureMapping[
+                  e.detail.feature_name]].samples.length;
+              const exampleIndices = e.detail.example_index == -1 ?
+                [...Array(this.data.length).keys()] : [e.detail.example_index];
+              for (let i = 0; i < numTrials; i++) {
+                for (let idx = 0; idx < exampleIndices.length; idx++) {
+                  const ex =
+                    JSON.parse(JSON.stringify(this.data[exampleIndices[idx]]));
+                  if (isNum) {
+                    ex.features.feature[
+                      e.detail.feature_name].floatList.value[0] =
+                      xMin + i * (1 / (numTrials - 1) *(xMax - xMin));
+                  } else {
+                    ex.features.feature[
+                      e.detail.feature_name].bytesList.value[0] =
+                      btoa(this.$.dash.partialDepPlotEligibleFeatures[
+                        featureMapping[e.detail.feature_name]].samples[i]);
+                  }
+                  examples.push(ex);
+                }
               }
-              examples.push(ex);
-            }
-            const tensors = [];
-            for (let i = 0; i < examples.length; i++) {
-              tensors.push(this.convertExToTensor(examples[i]));
-            }
-            const input = tf.concat(tensors).reshape([tensors.length, 105])
-            const res = this.model.predict(input, {batchSize: 128});
-            const predValues = await res.data();
-            const results = [];
-            for (let i = 0; i < examples.length; i++) {
-              const ex = examples[i];
-              const step = isNum
-                ? ex.features.feature[e.detail.feature_name].floatList.value[0]
-                : [atob(ex.features.feature[
-                    e.detail.feature_name].bytesList.value[0])];
-              const adjustedScore = 2 * this.means['age'] * predValues[i];
-              results.push({step: step, scalar: adjustedScore});
-            }
-            this.$.dash.makeChartForFeature_(isNum ? 'numeric' : 'categorical',
-                e.detail.feature_name, [{'value': results}]);
+              const results = [];
+              const predValuesList = [];
+              const BATCH_SIZE = 128
+              for (let i = 0; i < examples.length; i += BATCH_SIZE) {
+                let tlist = [];
+                for (let idx = i;
+                     idx < Math.min(i + BATCH_SIZE, examples.length);
+                     idx++) {
+                  const tensor =
+                  tlist.push(this.convertExToTensor(examples[idx]));
```

Comment:
idea: it would be an awesome addition if we can add `this.convertExamplesToTensors(iterator)` where the iterator returns an example to avoid some of these repetitions.

Reply:
i have a plan to consolidate the logic in the demos for less repeated code as it is getting to be a problem. was planning on saving it for a clean-up PR